scan/tls_handshake_scanners.go

```diff
@@ -0,0 +1,65 @@
+package scan
+
+import (
+	"errors"
+	"net"
+
+	"github.com/cloudflare/cfssl/scan/tls"
+)
+
+// TLSHandshakeScanners contains scanners testing host cipher suite negotiation
+var TLSHandshakeScanners = &Family{
+	Name:        "TLSHandshakeScanners",
+	Description: "Scans host's SSL/TLS version and cipher suite negotiation",
+	Scanners: []*Scanner{
+		{
+			"CipherSuiteScanner",
+			"Determines host's cipher suites accepted and prefered order",
+			cipherSuiteScan,
+		},
+	},
+}
+
+func sayHello(host string, ciphers []uint16) (serverCipher, serverVersion uint16, err error) {
+	tcpConn, err := net.Dial(Network, host)
+	if err != nil {
+		return
+	}
+	conn := tls.Client(tcpConn, &tls.Config{ServerName: host, CipherSuites: ciphers})
+	defer conn.Close()
+	serverCipher, serverVersion, err = conn.SayHello()
+	for _, cipherID := range ciphers {
+		if serverCipher == cipherID {
+			err = errors.New("Server negotiated ciphersuite we didn't send.")
```

Comment:
As a style note, Go error strings are generally not capitalised and don't end in punctuation.